### PR TITLE
Keras 中构建神经网络的 5 个步骤

### LINKED ISSUE #2322: 5 Step Life-Cycle for Neural Network Models in Keras
* 原文链接：[5 Step Life-Cycle for Neural Network Models in Keras](https://machinelearningmastery.com/5-step-life-cycle-neural-network-models-keras/)
  * Markdown文件：[文件地址](https://github.com/xitu/gold-miner/blob/master/TODO/5-step-life-cycle-neural-network-models-keras.md)
  * PR 地址：#2326
  * 文章分类：**机器学习**
  ----
  * 翻译时间：4 天
  * 校对时间：1 天
  * 翻译积分：4 分
  * 校对积分：1 分
  * [积分有什么用](https://github.com/xitu/gold-miner/wiki)

### PR DESCRIPTION
**Pull request by lsvih**
resolve #2322

### CODE REVIEW

**Comment by lsvih** [comment]
学习与使用 Keras 强烈推荐 [Keras 中文文档](http://keras-cn.readthedocs.io/en/latest/)，翻译的很到位，而且有各种样例。

**Comment by CACppuccino** [comment]
校对认领

**Comment by fanyijihua** [comment]
@CACppuccino 好的呢 :beer:

File: TODO/5-step-life-cycle-neural-network-models-keras.md [hunk 1]
```diff
@@ -3,87 +3,84 @@
 > * 原文作者：[Jason Brownlee](https://machinelearningmastery.com/author/jasonb/)
 > * 译文出自：[掘金翻译计划](https://github.com/xitu/gold-miner)
 > * 本文永久链接：[https://github.com/xitu/gold-miner/blob/master/TODO/5-step-life-cycle-neural-network-models-keras.md](https://github.com/xitu/gold-miner/blob/master/TODO/5-step-life-cycle-neural-network-models-keras.md)
-> * 译者：
+> * 译者：[lsvih](https://github.com/lsvih)
 > * 校对者：
 
-# 5 Step Life-Cycle for Neural Network Models in Keras
+# Keras 中构建神经网络的 5 个生命周期
```
Comment (CACppuccino) [inline TODO/5-step-life-cycle-neural-network-models-keras.md]: 不知意译为 Keras 中构建神经网络的 5 个步骤是否更直观一些

File: TODO/5-step-life-cycle-neural-network-models-keras.md [hunk 2]
```diff
@@ -3,87 +3,84 @@
...
 
-But we can also do this in one step by creating an array of layers and passing it to the constructor of the Sequential.
+此外，我们也可以通过创建一个层的数组，并将其传给 Sequential 构造器来定义模型。
 
 ```
 layers = [Dense(2)]
 model = Sequential(layers)
 ```
 
-The first layer in the network must define the number of inputs to expect. The way that this is specified can differ depending on the network type, but for a Multilayer Perceptron model this is specified by the input_dim attribute.
```
Comment (CACppuccino) [inline TODO/5-step-life-cycle-neural-network-models-keras.md]: [指定这个参数的方式有许多种] => [指定这个参数的方式有许多种，取决于要建造的模型种类]

File: TODO/5-step-life-cycle-neural-network-models-keras.md [hunk 3]
```diff
@@ -93,196 +90,181 @@ model.add(Dense(1))
...
 
-For example, below are some common predictive modeling problem types and the structure and standard activation function that you can use in the output layer:
+例如，以下是一些常用的预测建模问题类型，以及它们可以在输出层使用的结构和标准的激活函数：
 
-* **Regression**: Linear activation function or ‘linear’ and the number of neurons matching the number of outputs.
-* **Binary Classification (2 class)**: Logistic activation function or ‘sigmoid’ and one neuron the output layer.
-* **Multiclass Classification (>2 class)**: Softmax activation function or ‘softmax’ and one output neuron per class value, assuming a one-hot encoded output pattern.
+* **回归问题**：使用线性的激活函数 “linear”，并使用与与输出数量相匹配的神经元数量。
+* **二分类问题**：使用逻辑激活函数 “sigmoid”，在输出层仅设一个神经元。
+* **多分类问题**：使用 Softmax 激活函数 “softmax”；假如你使用的是 one-hot 编码的输出格式的话，那么每个输出对应一个神经元。
```
Comment (CACppuccino) [inline TODO/5-step-life-cycle-neural-network-models-keras.md]: 这里的第二个softmax是不是原作者的笔误，我在想他是不是想说对每一个分类用 sigmoid 函数搭配一个神经元

File: TODO/5-step-life-cycle-neural-network-models-keras.md [hunk 4]
```diff
@@ -93,196 +90,181 @@ model.add(Dense(1))
...
+在网络编译完成后，就能对它进行训练了。这个过程也可以看成是调整权重以拟合训练数据集。
 
-Fitting the network requires the training data to be specified, both a matrix of input patterns X and an array of matching output patterns y.
+训练网络需要制定训练数据，包括输入矩阵 X 以及相对应的输出 y。
 
-The network is trained using the backpropagation algorithm and optimized according to the optimization algorithm and loss function specified when compiling the model.
+在此步骤，将使用反向传播算法对网络进行训练，并使用在编译时制定的优化算法以及损失函数来进行优化。
 
-The backpropagation algorithm requires that the network be trained for a specified number of epochs or exposures to the training dataset.
+反向传播算法需要指定训练的 Epoch 数（轮次数、历元数）、对数据集的 exposure 数。
```
Comment (CACppuccino) [inline TODO/5-step-life-cycle-neural-network-models-keras.md]: Epoch => 回合数

File: TODO/5-step-life-cycle-neural-network-models-keras.md [hunk 5]
```diff
@@ -93,196 +90,181 @@ model.add(Dense(1))
...
 
-Each epoch can be partitioned into groups of input-output pattern pairs called batches. This define the number of patterns that the network is exposed to before the weights are updated within an epoch. It is also an efficiency optimization, ensuring that not too many input patterns are loaded into memory at a time.
+每个 epoch 都可以被划分成多组数据输入输出对，它们也称为 batch。batch 设定的数字将会定义在每个 epoch 中更新权重之前输入输出对的数量。这种做法也是一种优化效率的方式，可以确保不会同时加载过多的输入输出对到内存（显存）中。
```
Comment (CACppuccino) [inline TODO/5-step-life-cycle-neural-network-models-keras.md]: batch => batch(批量)

File: TODO/5-step-life-cycle-neural-network-models-keras.md [hunk 6]
```diff
@@ -93,196 +90,181 @@ model.add(Dense(1))
...
 ```
-# Sample Multilayer Perceptron Neural Network in Keras
+# Keras 多层感知机神经网络样例
 from keras.models import Sequential
 from keras.layers import Dense
 import numpy
-# load and prepare the dataset
+# 加载数据
 dataset = numpy.loadtxt("pima-indians-diabetes.csv", delimiter=",")
 X = dataset[:,0:8]
```
Comment (CACppuccino) [inline TODO/5-step-life-cycle-neural-network-models-keras.md]: 这里是有问题的，当时看教程就有些遗憾，后来发现标签和训练数据居然放到了一起

File: TODO/5-step-life-cycle-neural-network-models-keras.md [hunk 7]
```diff
@@ -93,196 +90,181 @@ model.add(Dense(1))
...
-accuracy = numpy.mean(predictions == Y)
-print("Prediction Accuracy: %.2f%%" % (accuracy*100))
+...
+768/768 [==============================] - 0s - loss: 0.5219 - acc: 0.7591
+Epoch 99/100
+768/768 [==============================] - 0s - loss: 0.5250 - acc: 0.7474
+Epoch 100/100
+768/768 [==============================] - 0s - loss: 0.5416 - acc: 0.7331
+32/768 [>.............................] - ETA: 0s
+Loss: 0.51, Accuracy: 74.87%
```
Comment (CACppuccino) [inline TODO/5-step-life-cycle-neural-network-models-keras.md]: 实际上不加入标签的话准确率只有69%左右

File: TODO/5-step-life-cycle-neural-network-models-keras.md (hunk 6 above)
Comment (lsvih) [reply TODO/5-step-life-cycle-neural-network-models-keras.md]: 我没找到这个.csv的数据集，希望了解一下什么情况，求微信商量一下

Comment (CACppuccino) [reply TODO/5-step-life-cycle-neural-network-models-keras.md]: 我对源码的理解有误，经沟通问题已解决。

**Comment by lsvih** [comment]
@leviding 已发布 https://juejin.im/post/59e43b5b6fb9a0452a3b5f4f
